packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -388,28 +427,149 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
+                    </div>
+                }
+                onConfirm={this.onBulkDeleteConfirm}
+                onCancel={this.onConfirmBulkCancel}
+                confirmVariant="danger"
+                confirmButtonText="Yes, Delete Fields"
+                cancelButtonText="Cancel"
+            />
+        );
+    }
+
+    onBulkDeleteClick = () => {
+        const { domain } = this.props;
+        const { visibleSelection } = this.state;
+        let fields = domain.fields;
+
+        let undeletableFields = [];
+        let deletableSelectedFields = [];
+
+        visibleSelection.forEach((val) => {
+            const field = fields.get(val);
+
+            if (field.isSaved() && !isFieldDeletable(field)) {
+                undeletableFields.push(val);
+            } else {
+                deletableSelectedFields.push(val);
+            }
+        });
+
+        const bulkDeleteConfirmInfo = {
+            deletableSelectedFields: deletableSelectedFields,
+            undeletableFields: undeletableFields,
+        }
+        this.setState({bulkDeleteConfirmInfo});
+    }
+
+    onBulkDeleteConfirm = () => {
+        const { domain } = this.props;
+        const { deletableSelectedFields } = this.state.bulkDeleteConfirmInfo;
+        const updatedDomain = removeFields(domain, deletableSelectedFields);
+        const visibleFieldsCount = getVisibleFieldCount(updatedDomain);
+        const visibleSelection = getVisibleSelectedFieldIndexes(updatedDomain.fields);
+        const rowIndexChanges = deletableSelectedFields.map((i) => {
+            return {originalIndex: i, newIndex: undefined } as DomainFieldIndexChange
+        });
+
+        this.onDomainChange(updatedDomain, true, rowIndexChanges);
+        this.setState({
+            visibleSelection,
+            visibleFieldsCount,
+            selectAll: visibleFieldsCount !== 0 && (visibleSelection.size === visibleFieldsCount),
+            bulkDeleteConfirmInfo: undefined,
+        }, () => {this.clearFilePreviewData(this.state)})
+    }
+
     onAddField = () => {
         this.applyAddField();
     };
 
     applyAddField = (config?: Partial<IDomainField>) => {
-        this.onDomainChange(addDomainField(this.props.domain, config));
+        const newDomain = addDomainField(this.props.domain, config);
+        this.onDomainChange(newDomain);
+        this.setState({selectAll: false, visibleFieldsCount: getVisibleFieldCount(newDomain)});
         this.collapseRow();
     };
 
+    // for ticking single
     onFieldsChange = (changes: List<IFieldChange>, index: number, expand: boolean) => {
-        this.onDomainChange(handleDomainUpdates(this.props.domain, changes));
-
+        const { domain } = this.props;
+        const { visibleFieldsCount } = this.state;
+        this.onDomainChange(handleDomainUpdates(domain, changes));
+
+        const firstChange = changes.get(0);
+        if (firstChange?.id.includes("domainpropertiesrow-selected")) {
```

Comment:
Not sure exactly where in the code this is happening but in a quick manual testing pass I saw that if I select/check a field's row checkbox, it updates the field details to say "Updated". Any way we can prevent this from being considered an "update" to the field?

Reply:
Good catch, fixed in function updateDomainField().